src/wms/src/test/java/org/geoserver/wms/map/png/GetMapIntegrationTest.java

```diff
@@ -146,4 +146,79 @@ public void testPng8Translucent() throws Exception {
             assertEquals(Transparency.TRANSLUCENT, cm.getTransparency());
         }
     }
+
+    @Test
+    public void testwms2_0_0Filter() throws Exception {
+        MockHttpServletResponse response =
+                getAsServletResponse(
+                        "wms?bbox="
+                                + bbox
+                                + "&styles=&layers="
+                                + layers
+                                + "&SERVICE=WMS&VERSION=1.1.1"
+                                + "&Format=image/png8"
+                                + "&request=GetMap"
+                                + "&width=550"
+                                + "&height=250"
+                                + "&srs=EPSG:4326&transparent=true"
+                                + "&FILTER=%3Cfes%3AFilter%20xmlns%3Axsi%3D%22http%3A%2F%2Fwww.w3.org%2F2001%2FXMLSchema-instance%22%20xmlns%3Agml%3D%22http%3A%2F%2Fwww.opengis.net%2Fgml%2F3.2%22%20xmlns%3Awfs%3D%22http%3A%2F%2Fwww.opengis.net%2Fwfs%22%20xmlns%3D%22http%3A%2F%2Fwww.opengis.net%2Ffes%2F2.0%22%20xmlns%3Afes%3D%22http%3A%2F%2Fwww.opengis.net%2Ffes%2F2.0%22%3E%3Cfes%3APropertyIsLike%20wildCard%3D%22*%22%20singleChar%3D%22.%22%20escapeChar%3D%22!%22%3E%3Cfes%3AValueReference%3ENAME%3C%2Ffes%3AValueReference%3E%3Cfes%3ALiteral%3E*United*%3C%2Ffes%3ALiteral%3E%3C%2Ffes%3APropertyIsLike%3E%3C%2Ffes%3AFilter%3E");
+        assertEquals("image/png; mode=8bit", response.getContentType());
+
+        try (InputStream is = getBinaryInputStream(response)) {
+            BufferedImage bi = ImageIO.read(is);
+            IndexColorModel cm = (IndexColorModel) bi.getColorModel();
+            assertEquals(Transparency.BITMASK, cm.getTransparency());
+            assertTrue(cm.getTransparentPixel() >= 0);
```

Comment:
The test is not checking if the filter got applied. A filter should reduce the number of features, so you should be able to check if a specific pixel in the output was not transparent before applying the filter, but it is now.

Also, as an addition, you should check the GetFeatureInfo JSON output, which makes for a easier to verify test: you verify features that were not matching the filter, are indeed not in the output.